packages/tailwindcss/src/candidate.test.ts

```diff
@@ -1416,3 +1416,130 @@ it('should parse candidates with a prefix', () => {
     ]
   `)
 })
+
+it.each([
+  // Empty arbitrary value
+  'bg-[]',
+  'bg-()',
+  //  Tricking the parser with a space is not allowed
+  'bg-[_]',
+  'bg-(_)',
+
+  // Empty arbitrary value, with typehint
+  'bg-[color:]',
+  'bg-(color:)',
+  //  Tricking the parser with a space is not allowed
+  'bg-[color:_]',
+  'bg-(color:_)',
+
+  // Empty arbitrary modifier
+  'bg-red-500/[]',
+  'bg-red-500/()',
+  //  Tricking the parser with a space is not allowed
+  'bg-red-500/[_]',
+  'bg-red-500/(_)',
+
+  // Empty arbitrary modifier for arbitrary properties
+  '[color:red]/[]',
+  '[color:red]/()',
+  //  Tricking the parser with a space is not allowed
+  '[color:red]/[_]',
+  '[color:red]/(_)',
+
+  // Empty arbitrary value and modifier
+  'bg-[]/[]',
+  'bg-()/[]',
+  'bg-[]/()',
+  'bg-()/()',
+  //  Tricking the parser with a space is not allowed
+  'bg-[_]/[]',
+  'bg-(_)/[]',
+  'bg-[_]/()',
+  'bg-(_)/()',
+  'bg-[]/[_]',
+  'bg-()/[_]',
+  'bg-[]/(_)',
+  'bg-()/(_)',
+  'bg-[_]/[_]',
+  'bg-(_)/[_]',
+  'bg-[_]/(_)',
+  'bg-(_)/(_)',
+
+  // Functional components
```

Comment:
React brain... this should say `Functional variants` and the other one `Compound variants` because I wanted to verify that we handled functional variants and compound variants correctly (since they are different types and handled separately).